python/cugraph/cugraph/gnn/data_loading/bulk_sampler_io.py

```diff
@@ -63,24 +67,73 @@ def _write_samples_to_parquet(
         if end_batch_id == max_batch_id:
             end_ix = len(results)
         else:
-            end_ix = offsets.offsets[offsets.batch_id == (end_batch_id + 1)].iloc[0]
+            offsets_z = offsets[offsets.batch_id == (end_batch_id + 1)]
+            end_ix = offsets_z.offsets.iloc[0]
 
         full_output_path = os.path.join(
             output_path, f"batch={start_batch_id}-{end_batch_id}.parquet"
         )
-        results_p = results.iloc[start_ix:end_ix]
+        results_p = results.iloc[start_ix:end_ix].reset_index(drop=True)
 
         results_p["batch_id"] = offsets_p.batch_id.repeat(
             cupy.diff(offsets_p.offsets.values, append=end_ix)
         ).values
-        results_p.to_parquet(full_output_path, compression=None, index=False)
+
+        if renumber_map is not None:
+            renumber_map_start_ix = offsets_p.renumber_map_offsets.iloc[0]
+
+            if end_batch_id == max_batch_id:
+                renumber_map_end_ix = len(renumber_map)
+            else:
+                renumber_map_end_ix = offsets_z.renumber_map_offsets.iloc[0]
+
+            renumber_map_p = renumber_map.map.iloc[
+                renumber_map_start_ix:renumber_map_end_ix
+            ]
+
+            # Add the length so no na-checking is required in the loading stage
+            map_offset = (
+                end_batch_id - start_batch_id + 2
+            ) - offsets_p.renumber_map_offsets.iloc[0]
+            renumber_map_o = cudf.concat(
+                [
+                    offsets_p.renumber_map_offsets + map_offset,
+                    cudf.Series(
+                        [len(renumber_map_p) + len(offsets_p) + 1], dtype="int32"
+                    ),
+                ]
+            )
+
+            renumber_offset_len = len(renumber_map_o)
+            if renumber_offset_len != end_batch_id - start_batch_id + 2:
+                raise ValueError("Invalid batch id or renumber map")
+
+            final_map_series = cudf.concat(
+                [
+                    renumber_map_o,
+                    renumber_map_p,
+                ],
+                ignore_index=True,
+            )
+
+            if len(final_map_series) > len(results_p):
+                # this should rarely happen and only occurs on small graphs/samples
+                final_map_series.name = "map"
+                results_p = results_p.join(final_map_series, how="outer").sort_index()
```

Comment:
We might get a perf hit because of this, lets add a todo here to fix it soon.

Reply:
ok